src/main/java/org/testng/reporters/FailedReporter.java

```diff
@@ -189,10 +162,8 @@ private void createXmlTest(ITestContext context, List<ITestNGMethod> methods, Xm
     Map<Class, Set<ITestNGMethod>> methodsMap= Maps.newHashMap();
 
     for (ITestNGMethod m : methods) {
-      Object[] instances= m.getInstances();
-      Class clazz= instances == null || instances.length == 0 || instances[0] == null
-          ? m.getRealClass()
-          : instances[0].getClass();
+      Object instances= m.getInstance();
```

Comment:
It is not related to this fix. But I changed this because 

* `org.testng.ITestNGMethod#getInstances` is deprecated.
* as per the javadocs of `org.testng.ITestNGMethod#getInstances`, the `getInstances()` method is deprecated and we should be using `getInstance()` instead of it.